Make linum toggle local and add a variable for global mode

StreakyCobra opened the pull request:
`global-linum-mode` is making line numbers appearing in `*spacemacs*`
and `helm` buffers, what is annoying.

This change make the linum `SPC t n` toggle buffer-local, and add a
customization variable in `.spacemacs` to enable line numbers globally.
«Globally» here as to be understood as «in `prog-mode` and `text-mode`».

robbyoconnor commented on doc/DOCUMENTATION.org:
```diff
@@ -936,6 +937,15 @@ and ~T~):
+
+#+begin_src emacs-lisp
+    (setq-default dotspacemacs-global-line-numbers t)
+#+end_src
```
Specify that this belongs in `dotpspacemacs/user-config` :stuck_out_tongue:

robbyoconnor replied:
This should also read: 
```
Line numbers can be toggled on in all =prog-mode= and =text-mode= buffers by setting the
```

TheBB commented on layers/+distribution/spacemacs-base/config.el:
```diff
@@ -165,6 +165,11 @@ It runs `tabulated-list-revert-hook', then calls `tabulated-list-print'."
+(if dotspacemacs-global-line-numbers
+    (progn
+      (add-hook 'prog-mode-hook 'linum-mode)
+      (add-hook 'text-mode-hook 'linum-mode)))
```
By the way, `(if cond (progn ...))` is equivalent to `(when cond ...)`

StreakyCobra commented:
@robbyoconnor I added the `buffers` word, but didn't specify the `dotpspacemacs/user-config` information as it shouldn't be placed there. It will be in `dotspacemacs/init`, and I copied the information message from similar variable [smooth-scrolling](https://github.com/syl20bnr/spacemacs/blob/develop/doc/DOCUMENTATION.org#smooth-scrolling), so it should be fine.

@TheBB Thx for the tip, updated :-)

robbyoconnor commented:
@StreakyCobra ah :)

TheBB commented:
Thank you! Love it :+1: Cherry-picked in develop. You can safely delete your branch.
